### ISSUE TITLE
SLOT lint false positive on classes with multiple ancestors

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#15879]
### Description

```
ruff check --select=SLOT000 .
```
```
ruff --version 0.8.4
```
```python
 class Backend(enum, str):
```
^ suggests a fixit here. This is the recommended class structure for old versions of python before strenum was introduced. However, enum doesn't have __slots__ defined, so defining a `__slots__` wouldn't be that helpful. i also see this bug happen when a class has multiple ancestors, one of which is a NamedTuple.

**Comment by MichaReiser** [comment]
Could you explain what's the bug experience? I don't think I understand it from the issue. 

Ruff does raise an error about a missing `__slots__` even if the base class is known. So warning in the case if any base class is unknown doesn't seem incorrect, because the `__slots__` attribute is always required if any parent is `tuple`. 

Playground: https://play.ruff.rs/45b205ae-153c-4a73-8fe6-29995ef0a5c9

**Comment by Skylion007** [comment]
@MichaReiser To clarify, what is the behavior if a class has two ancestors, and only one of them implements `__slots__`? My understanding is that the __slots__ in the class itself won't have an effect because it will still need to consult the `__dict__` of the first parent, but I could be wrong?

**Comment by MichaReiser** [comment]
Oh, I see. That's a good point. In that case, I'd say it's not a false positive, at least not when the base class is under the user's control. However, the suggested fix/message is misleading because only adding `__slots__` to the current class isn't enough; it also needs to be added to all non-tuple parent classes. 

Now, analyzing base classes is something that Ruff struggles with, at least if they're defined in another file. Which is why I'm leaning towards leaving the rule as is for now and revisit it once we have better type inference.
